plugins/modules/xdg_mime.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2025, Marcos Alano <marcoshalano@gmail.com>
+# Based on gio_mime module. Copyright (c) 2022, Alexei Znamensky <russoz@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: xdg_mime
+author:
+  - "Marcos Alano (@mhalano)"
+short_description: Set default handler for MIME type, for applications using XDG tools
+version_added: 10.6.0
+description:
+  - This module allows configuring the default handler for a specific MIME type, to be used by applications that use XDG.
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: full
+options:
+  mime_type:
+    description:
+      - MIME type for which a default handler will be set.
+    type: str
+    required: true
+  handler:
+    description:
+      - Default handler will be set for the MIME type.
+      - The desktop file must be installed in the system
+    type: str
+    required: true
+notes:
+  - This module is a thin wrapper around xdg-mime tool.
+  - See man xdg-mime(1) for more details.
+seealso:
+  - name: C(xdg-mime) command manual page
+    description: Manual page for the command.
+    link: https://portland.freedesktop.org/doc/xdg-mime.html
+  - name: xdg-utils Documentation
+    description: Reference documentation for xdg-utils.
+    link: https://www.freedesktop.org/wiki/Software/xdg-utils/
+"""
+
+EXAMPLES = r"""
+- name: Set chrome as the default handler for https
+  community.general.xdg_mime:
+    mime_type: x-scheme-handler/https
+    handler: google-chrome.desktop
+  register: result
+"""
+
+RETURN = r"""
+handler:
+  description:
+    - The handler set as default.
+  returned: success
+  type: str
+  sample: google-chrome.desktop
+stdout:
+  description:
+    - The output of the C(xdg-mime) command.
+  returned: success
+  type: str
+  sample: ''
+stderr:
+  description:
+    - The error output of the C(xdg-mime) command.
+  returned: failure
+  type: str
+  sample: ''
```

Comment:
Maybe something like `returned: never` since it won't return error. It sometimes returns, like when the handler has an invalid name, like `firefox.desktop2` (it should be just `.desktop`), but in other cases like a handler with `firefox2.desktop` that is a valid name but isn't present in the system, it won't return an error.

Reply:
In either case, there will be no error from Ansible, but the idempotency won't work.

Reply:
For some reason, probably the reason is I didn't test enough, it will return an error in case of a malformed handler. I will add to the documentation.